archunit/src/main/java/com/tngtech/archunit/core/domain/JavaType.java

```diff
@@ -15,289 +15,25 @@
-            }
-        }
-
-        private static String createPackage(String fullName) {
-            int packageEnd = fullName.lastIndexOf('.');
-            return packageEnd >= 0 ? fullName.substring(0, packageEnd) : "";
-        }
-
-        private static class PrimitiveType extends AbstractType {
-            PrimitiveType(String fullName) {
-                super(fullName, fullName, "");
-                checkArgument(primitiveClassesByName.containsKey(fullName), "'%s' must be a primitive name", fullName);
-            }
-
-            @Override
-            Class<?> classForName(ClassLoader classLoader) {
-                return primitiveClassesByName.get(getName());
-            }
-
-            @Override
-            public boolean isPrimitive() {
-                return true;
-            }
-
-            @Override
-            public JavaType withSimpleName(String simpleName) {
-                throw new UnsupportedOperationException("It should never make sense to override the simple type of a primitive");
-            }
-        }
-
-        private static class ArrayType extends AbstractType {
-            ArrayType(String fullName) {
-                this(fullName, createSimpleName(fullName), createPackageOfComponentType(fullName));
-            }
-
-            private ArrayType(String fullName, String simpleName, String packageName) {
-                super(fullName, simpleName, packageName);
-            }
-
-            private static String createPackageOfComponentType(String fullName) {
-                String componentType = getCanonicalName(fullName).replace("[]", "");
-                return createPackage(componentType);
-            }
-
-            private static String createSimpleName(String fullName) {
-                return ensureSimpleName(getCanonicalName(fullName));
-            }
-
-            private static String getCanonicalName(String fullName) {
-                return Type.getType(fullName).getClassName();
-            }
-
-            @Override
-            public boolean isArray() {
-                return true;
-            }
-
-            @Override
-            public JavaType withSimpleName(String simpleName) {
-                return new ArrayType(getName(), simpleName, getPackageName());
-            }
-
-            @Override
-            public Optional<JavaType> tryGetComponentType() {
-                String canonicalName = getCanonicalName(getName());
-                String componentTypeName = canonicalName.substring(0, canonicalName.lastIndexOf("["));
-                return Optional.of(JavaType.From.name(componentTypeName));
-            }
-        }
-    }
+import com.tngtech.archunit.PublicAPI;
+import com.tngtech.archunit.core.domain.properties.HasName;
+
+import static com.tngtech.archunit.PublicAPI.Usage.ACCESS;
+
+@PublicAPI(usage = ACCESS)
+public interface JavaType extends HasName {
+    /**
+     * Converts this {@link JavaType} into the erased type (compare the JLS).
```

Comment:
You refer to the Java Language Specification a few times in this PR.
IMO, it could be extremely helpful for the user to have direct links to the relevant chapter. 

If we could find a source whose URLs are considered stable enough (e.g. https://docs.oracle.com/javase/specs/jls/se8/html/ or https://docs.oracle.com/javase/specs/jls/se11/html/), I'd volunteer to contribute them, because in this way I'd actually check your statements. (Not that I wouldn't trust you, but you had asked for a review... 😉)

Reply:
Yeah, I didn't add a link because I wasn't sure how stable that would be :wink: But using the JLS 11 as the latest LTS version might make sense... I'll add https://docs.oracle.com/javase/specs/jls/se11/html/jls-4.html#jls-4.6

Reply:
On the other hand I just saw that we're using links to JLS 13 and 14 in other places and I remember that back then we just wanted to be "current" :disappointed:

Reply:
We should really use a consistent version I think, LTS would have the advantage that we don't have to change it too often, but on the other hand if you want to refer e.g. to the new records introduced in Java 14, then that makes it really hard :disappointed: Maybe we kind of need to stay at the current version always... But I can already see this to get outdated (as it already is), because noone will go through the code and update all links when a new Java version is released :thinking: Or maybe it's not that relevant as long as the link is not dead? :thinking: Those sections really don't change a lot between the releases and the concept always stays the same...

Reply:
I feel maybe we should use the latest LTS version to keep it stable. All the links/concepts at the moment also work if we link to the JLS 11. If we ever need a more modern concept (e.g. records for Java 14), we can just add a link for the current JLS. Then when a new LTS version is released we will just upgrade all the links to it in one shot (since then all concepts are present again...). That way we don't have to constantly adjust those links...

Reply:
I adjusted all the links to point to JLS version 11. Okay with you?

Reply:
Sure, thank you!